File: app/javascript/channels/activity_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ActivityChannel", {
  connected() {
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    const dot_element = document.getElementById(`user-${data['user_id']}-dot`);
    if(dot_element && data['online']){
      dot_element.classList.add('active');
    } else if (dot_element && !data['online']){
      dot_element.classList.remove('active');
    }
  }
});

File: app/channels/activity_channel.rb
class ActivityChannel < ApplicationCable::Channel
  def subscribed
    stream_from "activity_channel"
    redis.set("user_#{current_user.id}_online", "1")
    ActionCable.server.broadcast "activity_channel",
                                 { user_id: current_user.id.to_s,
                                 online: true }
  end

  def unsubscribed
    redis.del("user_#{current_user.id}_online")
    ActionCable.server.broadcast "activity_channel",
                                 { user_id: current_user.id.to_s,
                                 online: false }
  end

  private

  def redis
    Redis.new
  end
end
